### PR TITLE
🌱 Add GitHub workflow to scan project with CodeQL

### PR DESCRIPTION
**Pull request by camilamacedo86**
This workflow introduces CodeQL to automatically scan the project for security vulnerabilities, supporting efforts to maintain a secure codebase. It also helps us comply with standards like the Cybersecurity Certification Act (CCA) by ensuring continuous security monitoring.

Motivated by: https://github.com/kubernetes-sigs/kubebuilder/issues/3712

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Build project-v4 sample project
+        run: |
+          cd testdata/project-v4
+          go mod tidy
+          make all
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
```
Comment (grzesuav) [inline .github/workflows/codeql.yml:57]: can't go be hardcoded here ?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          build-mode: ${{ matrix.build-mode }}
+
+      - if: matrix.language == 'go' && matrix.build-mode == 'manual'
```
Comment (grzesuav) [inline .github/workflows/codeql.yml:60]: not sure when this condition is true (`manual` build mode)

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (camilamacedo86) [reply .github/workflows/codeql.yml:57]: Thx good point, I think we can simplify it !

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (camilamacedo86) [reply .github/workflows/codeql.yml:60]: Thx good point, I think we can simplify it !

**Comment by camilamacedo86** [comment]
Hi @grzesuav 

Really thank you for your review.
All addressed it is very simplified now.

So, since it is just a check that does not impact the end users and we addressed all I am moving forward
make no sense I make nobody waste so much time on reviews.

However, if you see any possible improvement here as anybody else please feel free to push a PR with your suggestions.

**Pull request merged**